common/ellipses.js

```diff
@@ -181,36 +181,56 @@
+                                scope.hideContent = true;
+                                containsOverflow = true;
+                                scope.maxHeightStyle =  maxHeightStyle;
+                            } else {
+                                scope.overflow[index] = false;
+                            }
+                        }
+
+                        // Resizerow is called whenever there is a change in rowValues model
+                        // It iterates over all the td elements and extracts image and iframes from it
+                        // After which it binds onload event to adjust height
+                        // It also calls updateHeight for the same td, for any oveflown textual content
                         var resizeRow = function() {
-                            if (containsOverflow == false && timerCount ++ < 500) {
+                            if (containsOverflow == false) {
 
+                                //Iterate over table data in the row
                                 for (var i = 0; i < element[0].children.length; i++) {
-                                    var height = element[0].children[i].children[0].clientHeight;
-                                    if (height > maxHeight) {
-                                        scope.overflow[i] = true;
-                                        scope.hideContent = true;
-                                        containsOverflow = true;
-                                        scope.maxHeightStyle =  maxHeightStyle;
-                                    } else {
-                                        scope.overflow[i] = false;
-                                    }
+
+                                    // Get all images and iframes inside the td  
+                                    var imagesAndIframes = UiUtils.getImageAndIframes(element[0].children[i]);
+
+                                    // Bind onload event and updateheight for particular td index
+                                    imagesAndIframes.forEach(function(el) {
+                                        var index = i;
+                                        el.onload = function() {
+                                            updateHeight(index, el);
+                                        };
+                                    });
+
+                                    updateHeight(i, element[0].children[i].children[0]);
                                 }
-                                $timeout(function() {
-                                    resizeRow();
-                                }, 50);
                             }
                         };
 
+                        // Watch for change in rowValues, this is useful in case of pagination
+                        // As Angular just changes the content and doesnot destroys elements
                         scope.$watchCollection('rowValues', function (v) {
```

Comment:
If I understand correctly, this `function (v)...` gets called when `rowValues` changes, which in turn delays `10` ms in case Angular (or the web browser layout engine?) is working on the layout, after which time calls the `resizeRow()` function. Is that right?

And what happens if the layout isn't settled down by the (10ms) time we call resizeRow, do we have a race condition possible or do we have a way of dealing with this possibility?